### PR TITLE
MEN-7570 - SP RBAC + ensured linter runs on all TS files

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

File: frontend/src/js/components/settings/role-management/DeleteRoleDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { Button, Dialog, DialogActions, DialogContent, DialogTitle } from '@mui/material';
+
+interface Props {
+  dismiss: () => void;
+  submit: () => void;
+  name: string;
+  open: boolean;
+}
+
+export const DeleteRoleDialog: FunctionComponent<Props> = ({ dismiss, open, submit, name }) => (
```
Comment (mineralsfree) [inline frontend/src/js/components/settings/role-management/DeleteRoleDialog.tsx]: Is it possible to use our `ConfirmModal` with some adjustments?

File: frontend/src/js/common-ui/forms/form.js [hunk 2]
```diff
@@ -41,6 +41,12 @@ const getErrorMsg = (validateMethod, args) => {
         return `Must be at least ${args[0]} characters long`;
       }
       break;
+    case 'isAlphanumericLocator':
+      if (args[0] && validator.isWhitelisted(args[0], 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_-')) {
```
Comment (mineralsfree) [inline frontend/src/js/common-ui/forms/form.js]: maybe 
```suggestion
      if (args[0] && validator.matches(args[0], /^[a-zA-Z0-9_-]+$/)) {
```
is a better way?

**Review by mineralsfree — DISMISSED** [review]
few remarks, + several new components are missing prop types, otherwise is good! :+1:

File: frontend/src/js/components/settings/role-management/DeleteRoleDialog.tsx (hunk 1 above)
Comment (mzedel) [reply frontend/src/js/components/settings/role-management/DeleteRoleDialog.tsx]: Good point! 🙊 if we don't tell anyone I think it should be fine 😄

**Review by mineralsfree — DISMISSED** [review]

**Review by mineralsfree — DISMISSED** [review]
Licenses file needs an update

**Review by mineralsfree — APPROVED** [review]

**Pull request merged**